charts/gateway-helm/templates/envoy-gateway-config.yaml

```diff
@@ -13,7 +13,15 @@ data:
         | quote }}
     kind: EnvoyGateway
     provider:
-      type: {{ .Values.config.envoyGateway.provider.type | quote }}
+      {{- toYaml .Values.config.envoyGateway.provider | nindent 6 }}
```

Comment:
My concern is everytime there's an new field introduced, you need update these lines too.
Is it possible render `Values.config.envoyGateway` directly?

Reply:
Hi @zirain, I have updated the manifest file to render the complete `values.config.envoyGateway` configuration and moved the logging levels to the values file as well.

Reply:
would you like to add an e2e test to ensure this work as expected?

Reply:
Sure, I would love to. 

WDYT about updating the current ratelimit tests manifests [here](https://github.com/envoyproxy/gateway/blob/main/examples/redis/redis.yaml#L52-L69) with a Kubernetes provider deployment spec(something like configuring cache key prefix) and checking the desired behavior in ratelimit e2e tests?

Reply:
you can reuse that manifests configuration, but in a sepreated test case.